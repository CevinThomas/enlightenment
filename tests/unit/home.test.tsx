import 'react-native';
import React from 'react';
import Home from '../../src/views/home';

import renderer from 'react-test-renderer';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

test('Home View Renders', () => {
    const tree = renderer.create(
        <Home/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
