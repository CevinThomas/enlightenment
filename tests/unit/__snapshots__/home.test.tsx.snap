// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "paddingTop": 30,
      "width": 675,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "century-gothic",
        "fontSize": 25,
      }
    }
  >
    Welcome future Beast!
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "elevation": 100,
        "marginTop": 20,
        "padding": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
          "lineHeight": 25,
        }
      }
    >
      This is place for you to be the fucking ultimate beast and advance your skills in the shit that is required. Please, pick a category.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          "SEO",
          "Javascript",
          "Python",
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 10,
                "color": "white",
                "flexDirection": "row",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 20,
                  "display": "flex",
                  "height": 150,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "borderRadius": 10,
                    "elevation": 100,
                    "height": "100%",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 26,
                      "height": "100%",
                      "lineHeight": 140,
                      "paddingTop": 5,
                      "textAlign": "center",
                      "width": "100%",
                      "zIndex": 9999,
                    }
                  }
                >
                  SEO
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 10,
                "color": "white",
                "flexDirection": "row",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 20,
                  "display": "flex",
                  "height": 150,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "borderRadius": 10,
                    "elevation": 100,
                    "height": "100%",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 26,
                      "height": "100%",
                      "lineHeight": 140,
                      "paddingTop": 5,
                      "textAlign": "center",
                      "width": "100%",
                      "zIndex": 9999,
                    }
                  }
                >
                  Javascript
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 10,
                "color": "white",
                "flexDirection": "row",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 20,
                  "display": "flex",
                  "height": 150,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "borderRadius": 10,
                    "elevation": 100,
                    "height": "100%",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 26,
                      "height": "100%",
                      "lineHeight": 140,
                      "paddingTop": 5,
                      "textAlign": "center",
                      "width": "100%",
                      "zIndex": 9999,
                    }
                  }
                >
                  Python
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "marginBottom": 20,
        "position": "absolute",
        "right": 0,
        "textAlign": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "overflow": "visible",
          "resizeMode": "contain",
          "width": 100,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "paddingTop": 30,
      "width": 675,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "century-gothic",
        "fontSize": 25,
      }
    }
  >
    Welcome future Beast!
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "elevation": 100,
        "marginTop": 20,
        "padding": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
          "lineHeight": 25,
        }
      }
    >
      This is place for you to be the fucking ultimate beast and advance your skills in the shit that is required. Please, pick a category.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          "SEO",
          "Javascript",
          "Python",
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 10,
                "color": "white",
                "flexDirection": "row",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 20,
                  "display": "flex",
                  "height": 150,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "borderRadius": 10,
                    "elevation": 100,
                    "height": "100%",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 26,
                      "height": "100%",
                      "lineHeight": 140,
                      "paddingTop": 5,
                      "textAlign": "center",
                      "width": "100%",
                      "zIndex": 9999,
                    }
                  }
                >
                  SEO
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 10,
                "color": "white",
                "flexDirection": "row",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 20,
                  "display": "flex",
                  "height": 150,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "borderRadius": 10,
                    "elevation": 100,
                    "height": "100%",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 26,
                      "height": "100%",
                      "lineHeight": 140,
                      "paddingTop": 5,
                      "textAlign": "center",
                      "width": "100%",
                      "zIndex": 9999,
                    }
                  }
                >
                  Javascript
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 10,
                "color": "white",
                "flexDirection": "row",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 20,
                  "display": "flex",
                  "height": 150,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "borderRadius": 10,
                    "elevation": 100,
                    "height": "100%",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 5,
                    "width": "100%",
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                      undefined,
                    ]
                  }
                />
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 26,
                      "height": "100%",
                      "lineHeight": 140,
                      "paddingTop": 5,
                      "textAlign": "center",
                      "width": "100%",
                      "zIndex": 9999,
                    }
                  }
                >
                  Python
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "marginBottom": 20,
        "position": "absolute",
        "right": 0,
        "textAlign": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "overflow": "visible",
          "resizeMode": "contain",
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
